SHS-6380: Spotlight Redesign

cienvaras opened the pull request:
# Summary
- Update spotlight component and related image styles, to apply new design

## Backstory
- ClickUp Tickets:
    - [Implementation: Spotlight Redesign](https://app.clickup.com/t/36718269/SHS-6380)
    - [Implement: new approach for long captions](https://app.clickup.com/t/36718269/SHS-6419)
    - [Implementation: Spotlight width](https://app.clickup.com/t/36718269/SHS-6390)

## Need Review By (Date)
11/12

## Urgency
medium

## Steps to Test

### Spotlight Redesign

1. In both [Traditional](https://hs-traditional-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com/) and [Colorful](https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com/) test sites, visit the following pages:
    1. Tall Spotlight(s) - Classic (`/components/hero-images-full-width-region-and-main-region/tall-spotlights-classic`)
    2. Short Spotlight(s) (`/components/hero-images-full-width-region-and-main-region/short-spotlights`
    3. Tall Spotlight(s) - Expanded (`/components/hero-images-full-width-region-and-main-region/tall-spotlights-expanded`)
    4. Short Spotlight(s) - Expanded (`/components/hero-images-full-width-region-and-main-region/tall-spotlights-expanded`)
2. For all spotlights, confirm that they match the new designs ([Traditional](https://www.figma.com/design/7xoXRIQgSMGlUszWYFbI09/SHS-Library---Traditional?node-id=1861-1039&t=NqgOzEOuXJpfPVt5-4), [Colorful](https://www.figma.com/design/UXl20gIVr8Vg4OmaTQBnYr/SHS-Library---COLORFUL?node-id=3893-3490&t=zoBaupMsH7WGf0hM-4))
3. Create a flexible page, add spotlights with the following characteristics (both in the full width and the main content area):
    1. Video with both right and left alignments
    2. No media
    3. Any other field combinations not tested before
4. Confirm that the spotlights match the designs in all cases
5. Edit or add a spotlight. Confirm that the "Image Style" field has the following help text: 
    > Framed images will show the same aspect ratio on all screens, but expanded images will more completely fill the screen. Be sure to set the image focal point and check multiple screen sizes to confirm that important parts of the image are not cut off. 

### Spotlight width

1. Log into the [Colorful Tugboat test site](https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/user)
2. Create a flexible page
3. Add a "Spotlight(s)" component in the Main Content region. Confirm that you can see the "Background Color Width" field on top of the component form. Set it to "Full Width".
4. Repeat the previous step, but set the "Background Color Width" field to "Limited Width".
5. Save the node. Confirm that the first Spotlight Slider background expands to the full page width, but the content stays centered.
6. Edit the node again. Add a "Spotlight(s)" component in the Full Width Region. Confirm that the "Background Color Width" field is not present.
7. Visit the [Tall Spotlight(s) - Classic](https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/components/hero-images-full-width-region-and-main-region/tall-spotlights-classic) page
8. Edit the node and confirm that all existing spotlight sliders in the Main Content Region have the "Background Color Width" field already set to "Limited Width". Change one of them to "Full width" and save the node.
9. Confirm that the Spotlight slider set to full width is unchanged (because the page have a sidebar).
10. Repeat the previous tests in the [Traditional test site](https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/user).

### new approach for long captions - spotlight specific adjustments

1. Visit [Tall Spotlight(s) - Expanded
](https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com/components/hero-images-full-width-region-and-main-region/tall-spotlights-expanded) and [Short Spotlight(s) - Expanded](https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com/components/hero-images-full-width-region-and-main-region/short-spotlights-expanded-0)
2. Confirm that the image captions in the spotlights look as expected and that there's no overlapping between the caption and the spotlight textbox.

# Review Tasks

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Are PHP functions and variables in `snake_case` and not `camelCase`?
- [ ] Does Drupal code follow [Drupal Coding Standards](https://www.drupal.org/docs/develop/standards/php/php-coding-standards)?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided?

### Code security
- [ ] Is all [user input properly sanitized when rendered](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

### General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1211869864871795

cienvaras committed:
feat(shs-6380): spotlight redesign

cienvaras committed:
feat(shs-6380): update spotlight's image style field help text

cienvaras committed:
Merge branch '11.16.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
feat(shs-6380): update spotlight image styles

cienvaras committed:
feat(shs-6380): spotlight component style improvements

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com
- https://hs-colorful-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com
- https://hs-traditional-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com
- https://english-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com
- https://history-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com
- https://west-fnxydosx1supxnmnowpf31cahvc05921.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/68d2d1f26f896278c142e4bd

Visual Diffs:
- https://dashboard.tugboatqa.com/68d2d1f26f896278c142e4bd/visualdiffs

cienvaras committed:
Merge branch '11.17.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
fix(shs-6380): sass linting issues

mariannuar committed:
Merge branch '11.18.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
chore(SHS-6380): start implementing new caption credit design

mariannuar committed:
chore(SHS-6380): start implementing new caption credit design

mariannuar committed:
chore(SHS-6419): implement new design for caption/credits

mariannuar committed:
chore(SHS-6419): implement new design for caption/credits

mariannuar committed:
chore(SHS-6419): update styles for caption/credits in spotlights

mariannuar committed:
chore(SHS-6419): update styles for caption/credits in spotlights

mariannuar commented:
Hi @cienvaras This ticket: https://app.clickup.com/t/36718269/SHS-6419 is ready for review on this branch.

cienvaras committed:
Merge branch '11.19.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+(function (Drupal) {
```
Use `once` library. That's important for preventing extra calls to the logic. You can use it to do initial changes and then register events on each of the captions.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+          const existingWrapper = caption.querySelector('.toggle-caption__content');
+
+          // Condition to add toggle
+          const addToggle = actualHeight >= 38 || (parentSpotlight && viewportWidth <= 991);
```
Use `matchMedia()` for viewport media queries.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+            caption.classList.add('toggle-caption__wrapper');
+
+            // Create wrapper
+            const textWrapper = document.createElement('div');
```
No need to add this `div` with Javascript, it can be added in the template to prevent extra processing in JS. You can control styles and visibility with a class or a data attribute that gets added when the element spans more than 2 lines.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+            // Create wrapper
+            const textWrapper = document.createElement('div');
+            textWrapper.className = 'toggle-caption__content';
+            textWrapper.style.visibility = 'hidden';
```
Visibility hidden will also hide the text for screen readers.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+            caption.appendChild(textWrapper);
+
+            // Create close button (inside wrapper, before text)
+            const closeBtn = document.createElement('button');
```
I'd explore using a single "toggle" button instead of two different ones. That might simplify the code a bit.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/scss/partials/utilities/_caption-credit.scss:
```diff
@@ -61,6 +62,64 @@
+      .toggle-caption__close {
+        background: none;
+        border: none;
+        float: left;
```
If positioned right inside the markup, no need for this `float` property.

cienvaras commented on docroot/themes/humsci/humsci_basic/src/js/shared/media/media-caption-toggle.js:
```diff
@@ -0,0 +1,103 @@
+
+      // Run again on viewport resize
+      let resizeTimeout;
+      window.addEventListener('resize', () => {
```
Explore using [ResizeObserver](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) for the individual captions instead of a global resize event.

mariannuar replied:
I tried to use it but there was something causing a loop on captions that were longer in large screensizes. I couldn't find what and why it was happening. And it was specifically with this function.

mariannuar replied:
I tried to implement it by changing the logic it was getting more complicated than expected. `ResizeObserver` observes the rendered size of an element (offsetHeight, contentRect, etc.). By adding or removing classes (like `visually-hidden` or `collapsible-caption`) it changes the layout of the element, and each layout change triggers the observer again.

Example:
- `updateCaptionState` adds `visually-hidden` because the caption is collapsible.
- The browser recalculates layout, so ResizeObserver fires again.
- `updateCaptionState` runs again, it add/remove classes again.
- This creates an infinite loop, especially because offsetHeight keeps changing slightly due to rounding or wrapping.

cienvaras reviewed (commented):
Partial review for [SHS-6419: Implement: new approach for long captions](https://app.clickup.com/t/36718269/SHS-6419). Please see inline comments.

mariannuar committed:
chore(SHS-6419): update logic for long captions for better optimization

mariannuar committed:
chore(SHS-6419): update logic for long captions for better optimization

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

mariannuar commented:
@cienvaras Thank you for your feedback! Hopefully this approach looks and works much better. Please let me know if there's something missing. This is ready for another review

cienvaras committed:
Merge branch '11.19.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
chore(SHS-6419): update logic for toggle caption - make it work better

mariannuar committed:
chore(SHS-6419): update logic for toggle caption - make it work better

mariannuar committed:
chore(SHS-6419): add condition for expanded spotlight in colorful only since design it's different

mariannuar committed:
chore(SHS-6419): add condition for expanded spotlight in colorful only since design it's different

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

mariannuar committed:
chore(SHS-6419): fix padding of toggle in traditional and update style in banner image with text box in colorful

mariannuar committed:
chore(SHS-6419): fix padding of toggle in traditional and update style in banner image with text box in colorful

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

mariannuar commented:
Hi @cienvaras ! This should be ready for another review. I left you a comment above related to `ResizeObserver ` and here's a [loom video](https://www.loom.com/share/e562901688d94017892cbebb2c0b875f) where it shows that there shouldn't be a delay when updating the caption

cienvaras committed:
Merge branch '11.19.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
chore(SHS-6419): add styles when image is on the left

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

mariannuar committed:
chore(SHS-6419): update logic

mariannuar committed:
chore(SHS-6419): update logic

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

mariannuar committed:
chore(shs-6380): remove extra scss

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

cienvaras committed:
SHS-6390: Implementation  Spotlight width (#2039)

* feat(shs-6390): add background color width field to the spotlight slider
paragraph type

* feat(shs-6380): hide background color width field when spotlight slider
is in the full width region

* feat(shs-6380): support full width spotlight sliders in the main content
region

* feat(shs-6390): update spotlight slider display settings

* feat(shs-6390): update styles to support full width spotlights in main
content area

* feat(shs-6390): add update hooks to set default value for new
field_bg_color_width in spotlight sliders

---------

Co-authored-by: mariannuar <mariannuar@gmail.com>

cienvaras committed:
fix(shs-6380): minor fixes in updated caption styles and js logic

cienvaras committed:
fix(shs-6380): minor fixes in updated caption styles and js logic

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

mariannuar committed:
chore(SHS-6419): update max-width to avoid long captions overlaping with text box in banner images

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

cienvaras commented:
@ahughes3 @Jnwibowo7321 This one is ready for review. Please remember this PR includes 3 related tickets.

Jnwibowo7321 commented:
Hello @cienvaras 
I'm looking at this ticket https://app.clickup.com/t/36718269/SHS-6380
and the new spotlight design, just want to confirm with youthe spotlight have more all around padding is that right?
<img width="719" height="389" alt="image" src="https://github.com/user-attachments/assets/4bdd1d5f-5285-4089-ad37-3c3d0dcb5563" />
<img width="720" height="379" alt="image" src="https://github.com/user-attachments/assets/ec3e5af0-a12e-41b7-89bf-ee2dd5be9fa4" />

Left is the tugboat right is the prod version

https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/components/hero-images-full-width-region-and-main-region/short-spotlights-classic
vs. https://hs-colorful.stanford.edu/components/hero-images-full-width-region-and-main-region/short-spotlights-classic

Short spotlight seems odd with the big padding

I've confirmed with Tori that it does not look correct

cienvaras committed:
Merge branch '11.20.2-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
TEMP

cienvaras committed:
chore(shs-6419): remove long caption related styles from spotlight

cienvaras committed:
fix(shs-6419): minor fixes in image caption/credit styles

cienvaras committed:
fix(shs-6419): incorrect dimensions in media-caption-toggle script on traditional

cienvaras committed:
Merge branch '11.21.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
fix(shs-6380): cleanup and update spotlight component styles

cienvaras committed:
Merge branch '11.21.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
Merge branch '11.21.1-release' of github.com:SU-HSDO/suhumsci into shs-6419--new-approach-long-captions

cienvaras committed:
Merge branch '11.21.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://english-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://history-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com
- https://west-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0

Visual Diffs:
- https://dashboard.tugboatqa.com/6902a2dba19aabe9595a68a0/visualdiffs

cienvaras committed:
Merge branch '11.22.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

cienvaras committed:
fix(shs-6380): update spotlight image styles to match layout changes

cienvaras committed:
fix(shs-6380): minor fix in spotlight image styles

cienvaras commented:
@Jnwibowo7321 @ahughes3 This one is ready for testing. As discussed last week, please ignore the image captions for now, because we're still for final design and implementation.

To make things easier, I created a test pages that includes other spotlight types (video, text only, etc.) that are not included in the default spotlight test pages:

- Colorful: https://hs-colorful-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/spotlight-test
- Traditional: https://hs-traditional-6qepuqsegwq15iieclzcklidaebb1z8e.tugboatqa.com/spotlight-test

Jnwibowo7321 reviewed (approved):
LGTM, please dont merge until the prod release of 2/11

mariannuar committed:
Merge branch '11.24.1-release' of github.com:SU-HSDO/suhumsci into shs-6419--new-approach-long-captions

joegl commented on docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.install:
```diff
@@ -656,3 +656,53 @@ function hs_paragraph_types_update_10017() {
+}
+
+/**
+ * Set default value for new heading level field in existing accordions.
```
There are multiple comments for this update referencing accordion paragraphs. These should reflect the spotlight slider changes being made here.

joegl commented on docroot/modules/humsci/hs_layouts/hs_layouts.module:
```diff
@@ -159,24 +159,49 @@ function hs_layouts_preprocess_container(&$variables) {
-    $className = $paragraph->get('field_spotlight_style')->isEmpty() ? 'classic' : $paragraph->get('field_spotlight_style')->getString();
-    $variables['attributes']->addClass(Html::cleanCssIdentifier('hb-spotlight--' . $className));
+
+  $spotlight_type = NULL;
```
Why not just set this to "classic" here if that's always going to be the default?

joegl reviewed (changes requested):
Looks good just a few minor suggestions.

mariannuar committed:
chore(shs-6523): implement new styles for captions

mariannuar committed:
chore(shs-6523): add styles for images with caption on text area or nodes

cienvaras commented:
@ahughes3 @Jnwibowo7321 Just a heads up, we still need to update the captions before this one is ready for review. I was assigned to Joey from a previous iteration but forgot to re-assigned. We'll let you know when it's ready for review again.

mariannuar committed:
fix(shs-6523): caption styles on vertical linked cards

mariannuar committed:
fix(shs-6523): caption styles on heros

mariannuar committed:
fix(shs-6523): lint issues

mariannuar committed:
Merge branch '11.24.1-release' of github.com:SU-HSDO/suhumsci into shs-6419--new-approach-long-captions

mariannuar committed:
Merge branch '11.24.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
fix(shs-6480): make comments more accurate and better code

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
chore(shs-6522): update new Spotlight with new captions design

mariannuar committed:
fix(shs-6380): php code

mariannuar committed:
fix(shs-6380): php code

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-traditional-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://english-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://history-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://west-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548

Visual Diffs:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548/visualdiffs

mariannuar commented:
@cienvaras Ready for another review!

cienvaras committed:
chore(shs-6419): remove trimming config on media caption field

cienvaras commented on docroot/themes/humsci/humsci_basic/src/scss/partials/components/_spotlight-slider.scss:
```diff
@@ -54,6 +48,19 @@
   }
+
+  &:has(.hb-spotlight--expanded):has(.field-media-image-caption) {
+    @include grid-media-min('md') {
```
@mariannuar Why are you combining regular media queries with content queries? I think it's better to use only one.

mariannuar replied:
@cienvaras I was trying to use some content queries logic here because we need to update the bottom only on screen sizes above 768px. Without this, the arrows' position would change in all the small screen sizes.

cienvaras replied:
@mariannuar That makes sense!

cienvaras reviewed (commented):
@mariannuar Please take a look at this

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6419--new-approach-long-captions

mariannuar committed:
fix(shs-6419): markup and update caption name class to a better one

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar committed:
fix(shs-6419): test for captions

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-traditional-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://english-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://history-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://west-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548

Visual Diffs:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548/visualdiffs

cienvaras commented:
@ahughes3 @Jnwibowo7321 Ready for review.

Please wait for #2082 to be merged before merging this one. Otherwise, we'll we be merging the site wide changes to the captions early and also adding some possible conflicts that might be difficult to fix.

mariannuar committed:
feat(shs-6419): use figcaption tag instead

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-traditional-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://english-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://history-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://west-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548

Visual Diffs:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548/visualdiffs

mariannuar committed:
chore(shs-6419): revert changes on test

mariannuar committed:
Merge branch 'shs-6419--new-approach-long-captions' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

ahughes3 commented:
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Links:
- https://pr1979-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-colorful-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://hs-traditional-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://english-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://history-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com
- https://west-t28udft8krytqgumtb9umgov0q0g9qay.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548

Visual Diffs:
- https://dashboard.tugboatqa.com/698f649d017fd55f32d66548/visualdiffs

Jnwibowo7321 reviewed (approved):
LGTM on tugboat

joegl commented:
@cienvaras Looks like there's a minor conflict in the js file here. Let me know once you get it fixed up and we should be able to get this in 👍

mariannuar committed:
Merge branch '11.24.1-release' of github.com:SU-HSDO/suhumsci into shs-6380--spotlight-redesign

mariannuar commented:
@joegl Ready!
